### DISCUSSION TITLE
OPA CLI EVAL

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#378] [OPA and Rego]
Hi there,

I'm currently reading https://www.openpolicyagent.org/docs/latest/cli/#opa-eval, could you confirm if I can evaluate the entire package by using 'data.<package>'?

In Visual studio code, through OPA plug-in 'OPA: Evaluate Package', I'm able to evaluate entire package. I'm just wondering if one CLI command can do the same.

Additionally can OPA rest API evaluate the entire rego package?

Thanks!

**Comment by anderseknert** [comment]
Yes, you can evaluate "everything" by querying `data` both via the CLI and the REST API.

```
opa eval -d some_policy.rego -d some_data.json data
```

```
curl localhost:8181/v1/data
```

**Reply by anderseknert** [reply]
See https://www.openpolicyagent.org/docs/latest/philosophy/#the-opa-document-model for a good read on the topic.

**Reply by Ronnie-personal** [reply]
Thanks for sharing the document! I have reviewed.

I need your help to understand 'Entitlements data pulled asynchronously and located under data.entitlements'.

Is entitlements the JSON file name or the Rego file name?

Thanks!
